### ISSUE TITLE
Version 0.32.0 feedback

### ISSUE THREAD

**Issue by peter-conalgo** [josiahcarlson/rom#69]
Tried out the new version. Thanks for making changes!

Here's some feedback/surprises. I'm coming from a recent SQLAlchemy project BTW.

- `Model.query.all()` doesn't work as I'd expect. To me it should be equivalent to `list(Model.query)`
- Same comment for `Model.query.first()` but if it's not ordered, then maybe not.
- If I have a sparse table, perhaps the next id in sequence is 10000, but there are only 3 rows. I think `Query._iter_all()` would be very slow in this case. Perhaps a bit of Lua to do HKEYS and copy that into a temp list? Would be easy to sort that on server-side, if desired.

To my thinking, `Model.query` with no filtering means I expect to operate over the primary key. If it's cheap, it should be ordered by the id, for my convenience/laziness. Although you don't have a sorted-set for the id index, you can build a list from HKEYS>SORT>store. Then remaining query operations would just leverage existing code for indexed columns using that temporary list.

**Comment by josiahcarlson** [comment]
Yeah, I forgot about those query options because I don't use empty queries against rom. I'll be releasing 0.32.1 shortly with some changes.

Regarding the HKEYS -> SORT -> store - I didn't want to do that because I didn't want to induce a bunch of overhead for something that could be done incrementally. Thinking about it a little more, to keep the incremental operations while also ensuring no duplicates, I ended up writing a Lua script that scans the hash using HSCAN, and returns the full entity data; the next call to the Lua script records the ids into a SET, then does the HSCAN + fetch the data.

Other stuff:
* I made `Model.query.all()` work (it does `list(self)` in the context of the query)
* I made `Model.query.first()` work (it does `for ent in self: return ent` in the context of the query)
* I added an optimized version of the `Query.iter_result()` (aka `__iter__`) for the case where there exists an index on the `PrimaryKey()` column, and the aforementioned HSCAN optimization when using Redis 2.8+

I'm going to let this bake in my head for the evening, likely releasing 0.32.1 tomorrow evening.
